sql/async_udfs.mdx

```diff
@@ -0,0 +1,117 @@
+---
+title: Asynchronous User-Defined Functions
+description: 'Run Async UDFs'
+---
+
+Arroyo's UDFs can be asynchronous.
+This is useful for UDFs that need to make network requests or perform other asynchronous operations,
+and therefore should not block processing. Since asynchronous UDFs are executed in parallel,
+we can maintain high throughput even with longer-running UDFs.
+
+Note that async UDFs should not block (i.e. `await`) as this will block the main processing thread.
```

Comment:
This is phrased in a way that might be confusing for users.

In Rust async code, you must use async primitives (i..e, things you `await` on) rather than blocking code. I.e., `std::time::sleep(..)` is bad because it blocks, but `tokio::time::sleep(..).await` is good because it hooks into the async machinery to defer processing.